File: win-prog/ProgWinCS/Text and Fonts/GetFamiliesList/GetFamiliesList.cs
//----------------------------------------------
// GetFamiliesList.cs © 2001 by Charles Petzold
//----------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class GetFamiliesList: FamiliesList
{
     public new static void Main()
     {
          Application.Run(new GetFamiliesList());
     }
     public GetFamiliesList()
     {
          Text = "Font GetFamilies List";
     }
     protected override FontFamily[] GetFontFamilyArray(Graphics grfx)
     {
          return FontFamily.GetFamilies(grfx);
     }
}

File: win-prog/ProgWinVB/Text and Fonts/GetFamiliesList/GetFamiliesList.vb
'------------------------------------------------
' GetFamiliesList.vb (c) 2002 by Charles Petzold
'------------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class GetFamiliesList
    Inherits FamiliesList

    Shared Shadows Sub Main()
        Application.Run(New GetFamiliesList())
    End Sub

    Sub New()
        Text = "Font GetFamilies List"
    End Sub

    Protected Overrides Function GetFontFamilyArray _
                                (ByVal grfx As Graphics) As FontFamily()
        Return FontFamily.GetFamilies(grfx)
    End Function
End Class
